Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -9025,7 +9032,11 @@ void WebPageProxy::createNewPage(IPC::Connection& connection, WindowFeatures&& w
         if (privateClickMeasurement)
             newPage->internals().privateClickMeasurement = { { WTFMove(*privateClickMeasurement), { }, { } } };
 
-        if (navigationDataForNewProcess && !openedBlobURL) {
+        // Blobs get a dedicated process when site isolation is enabled and the navigation is cross-site. This is because blobs will lose access to the opener process.
+        bool isCrossSiteNavigation = Site(mainFrameURL) != Site(request.url());
+        bool blobRequiresNewProcess = openedBlobURL && protectedPreferences()->siteIsolationEnabled() && isCrossSiteNavigation;
+
+        if (navigationDataForNewProcess && (blobRequiresNewProcess || !openedBlobURL)) {
```

Comment:
Why not just `if (navigationDataForNewProcess && (protectedPreferences()->siteIsolationEnabled() || !openedBlobURL)) {`?